python/pyspark/sql/types.py

```diff
@@ -1756,6 +1854,15 @@ def _parse_datatype_json_value(json_value: Union[dict, str]) -> DataType:
             )
 
 
+def _parse_type_with_collation(
+    json_value: Dict[str, Any], fieldPath: str, collationsMap: Optional[Dict[str, str]]
+) -> DataType:
```

Comment:
Don't we also need the string type check here?

Reply:
yes, added now!